src/engine/core/fs/tcp_file_device.cpp

```diff
@@ -67,7 +67,7 @@ namespace Lumix
 				MT::SpinLock lock(m_spin_mutex);
 				m_stream->write(op);
 				m_stream->write(m_file);
-				m_stream->write(size);
+				m_stream->write((uint64)size);
```

Comment:
Nice catch, I've created an issue #824. I am thinking about deleting this whole tcp thing, it's not used anywhere at the moment anyway, so there is no easy way to test it.